js/router.js

```diff
@@ -159,43 +184,107 @@ module.exports = Backbone.Router.extend({
     window.obEventBus.trigger('cleanNav');
   },
 
-  newView: function(view, bodyID, addressBarText, bodyClass){
-    var loadingConfig;
+  cacheView: function(view, fragment) {
+    var index;
+
+    fragment = fragment || Backbone.history.getFragment();
+    index = view.constructor.getCacheIndex(fragment);
+    
+    this.viewCache[index] = {
+      cachedAt: Date.now(),
+      view: this.view
+    }
+  },
+
+  newView: function(View, options) {
+    var now = Date.now(),
+        cached = this.viewCache[View.getCacheIndex(Backbone.history.getFragment())],
+        requestedRoute = Backbone.history.getFragment(),
+        loadingConfig;
+
+    options = __.extend({
+      // viewArgs can be an array of args to pass into the view or a single
+      // arg (most likely an options object)
+      viewArgs: [{}],
+      addressBarText: '',
+      bodyID: '',
+      bodyClass: ''
+    }, options || {});
+
+    options.viewArgs = options.viewArgs.length ? options.viewArgs : [options.viewArgs];
 
     this.cleanup();
-    //clear address bar. This will be replaced on the user page
-    addressBarText = addressBarText || "";
-    window.obEventBus.trigger("setAddressBar", addressBarText);
+    window.obEventBus.trigger('setAddressBar', options.addressBarText);
 
-    if ($('body').attr('id') != bodyID){
-      $('body').attr("id", bodyID || "");
-    }
-    if (bodyClass){
-      $('body').attr('class', bodyClass);
-    }
-    $('#obContainer').removeClass("customizeUserPage"); //remove customization styling if present
+    $('body').attr('id', options.bodyID);
+    $('body').attr('class', options.bodyClass);
+    $('#obContainer').removeClass('customizeUserPage'); //remove customization styling if present
     
     this.pageConnectModal && this.pageConnectModal.remove();
     this.pageConnectModal = null;
 
-    if (
-      (loadingConfig = __.result(view, 'loadingConfig')) &&
-      loadingConfig.promise &&
-      typeof loadingConfig.promise.then === 'function') {
-      this.launchPageConnectModal(loadingConfig);
+    // let's update the cache of our existing view (if it's cached && not expired) so
+    // cachedAt is updated and the user has up until the view's 'cacheExpires' amount of
+    // time to come back to it in it's current state.
+    for (var key in this.viewCache) {
+      let cached = this.viewCache[key],
+          curHash = location.hash.slice(1);
```

Comment:
No idea - I'm sure it had a use at some point, but I must have changed the code and forgot to clean it up. I've removed it.